controller/Booklet.php

```diff
@@ -187,72 +198,80 @@ public function delete()
     /**
      * Creates new instance of booklet
      * @return mixed
-     * @throws \common_ext_ExtensionException
+     * @throws common_ext_ExtensionException
      */
     public function wizard()
     {
         $this->defaultData();
 
         try {
-            $bookletClass  = $this->getCurrentClass();
-            $formContainer = new WizardBookletForm($bookletClass);
-            $myForm        = $formContainer->getForm();
-
-            if ($myForm->isSubmited()) {
-                if ($myForm->isValid()) {
-                    $test = $this->getResource($myForm->getValue(tao_helpers_Uri::encode(BookletClassService::PROPERTY_TEST)));
-
-                    return $this->returnTaskJson(CompileBooklet::createTask(
-                        $test,
-                        $bookletClass,
-                        $myForm->getValues()
-                    ));
-                } else {
-                    return $this->returnJsonError(__('Fill in all required fields'));
+            $form = (new WizardBookletForm($this->getCurrentClass()))->getForm();
+
+            if ($form !== null && $form->isSubmited()) {
+                if ($form->isValid()) {
+                    $test = $this->getResource(
+                        $form->getValue(tao_helpers_Uri::encode(BookletClassService::PROPERTY_TEST))
+                    );
+
+                    return $this->returnTaskJson(
+                        CompileBooklet::createTask($this->getCurrentClass(), $test, $form->getValues())
+                    );
                 }
+
+                return $this->returnJsonError(__('Fill in all required fields'));
             }
```

Comment:
Consider reducing the nesting level of the conditional statements with something like the following piece of code.

```suggestion
            if ($form === null || !$form->isSubmited()) {
                $this->renderForm($form);

                return;
            }

            if (!$form->isValid()) {
                return $this->returnJsonError(__('Fill in all required fields'));
            }

            $test = $this->getResource(
                $form->getValue(tao_helpers_Uri::encode(BookletClassService::PROPERTY_TEST))
            );

            return $this->returnTaskJson(
                CompileBooklet::createTask($this->getCurrentClass(), $test, $form->getValues())
            );
```